srv/salt/_modules/osd.py

```diff
@@ -898,8 +898,10 @@ def _bluestore_partitions(self):
                     log.warning(("WAL size is unsupported for same device of "
                                  "{}".format(self.osd.device)))
                 else:
-                    # Create wal of wal_size on wal device
-                    self.create(self.osd.wal, [('wal', self.osd.wal_size)])
+                    log.info(("WAL will reside on same device {} as db - "
+                              "recommend removing the WAL entry from the "
+                              "configuration for device "
+                              "{}").format(self.osd.db, self.osd.device))
             else:
                 # pylint: disable=line-too-long
                 log.warning("No size specified for wal {}. Using default sizes.".format(self.osd.wal))
```

Comment:
Unrelated to this PR but just in case - are this line and similar handling for absent db_size relevant? It looks like respective devices wouldn't be created at all rather than "using default sizes".

Reply:
If a size is not specified, we cannot create a partition.  However, ceph-disk will accept a device without the partition and create partitions itself.  So, the "default sizes" are for what ceph-disk will use.